firedrake/variational_solver.py

```diff
@@ -225,8 +237,10 @@ def solve(self, bounds=None):
         # Make sure appcontext is attached to the DM before we solve.
         dm = self.snes.getDM()
         # Apply the boundary conditions to the initial guess.
-        for bc in self._problem.bcs:
-            bc.apply(self._problem.u)
+        from firedrake.bcs import DirichletBC
+        for b in self._problem:
+            if isinstance(b, DirichletBC):
+                b.apply(self._problem.u)
```

Comment:
So I would rather than `NLVProblem` were not iterable, and this were just `for bc in chain(*self._problem.bcs):`